HULKs/hulk
# GUI for manual calibration

## Introduced Changes

**REQUIRES** #184

Provides a GUI for manual calibration. In addition, this incorporates the save-to-disk facility from the parameters panel. For detailed ideas and the implementation plans, please take a look at #117 .

### Actual files changes
Due to dependency on above mentioned #184 , a large diff is shown. Below are the files actually added/ changed in this PR.
```bash
$ gd 121_locate_configuration_files_twix 117_gui_for_manual_calibration --name-only
tools/twix/src/main.rs
tools/twix/src/panels/manual_camera_calibration.rs
tools/twix/src/panels/mod.rs
tools/twix/src/panels/parameter.rs
```

Fixes #

#117 

## ToDo / Known Issues

- [x] UI stuff probably is ultra* hacky and bad, fix them.
    - While not the best, esp in UX aspects, this is now in a useable state. Tested on Nao and Webots.
- [x] Add sliders instead of editors
- [x] Single save button for all? or one per parameter (or one for top, one for bottom)?
    - Split per camera.
- [x] slider limit calculation
    - Decided on +- 15 deg for extrinsic. This is a good value based on past experience.

## Ideas for Next Iterations (Not This PR)

*If there are some improvements that could be done in a next iteration, describe them here.*

## How to Test

![image](https://user-images.githubusercontent.com/6742239/233802147-1efddf0d-fed7-4041-bc84-3978816d9455.png)

Open twix, load the panel "Manual Calibration", there the UI should present calibration ability for intrinsic & extrinsic values.
The UI should be intuitive enough to figure out the rest, use the sliders to adjust parameters. Use save to disk to save on the PC with the correct head/ body ID mapping (Uses feature implemented in #184 )
